### ISSUE TITLE
Serialization exception when assigning AttrList to a field with multi=True

### ISSUE THREAD

**Issue by AbelThorne** [elastic/elasticsearch-dsl-py#748]
```
class TestDoc(DocType):
        my_field = Keyword(multi=True)

        def my_func(self):
            # This fails
            self.my_field = AttrList(["value1", "value2", "value3"])
            # This works
            # self.my_field = [v for v in AttrList(["value1", "value2", "value3"])]

    myTest = TestDoc()
    myTest.my_func()
    myTest.save(index="my_index")
```

The above code fails with a SerializationError when saving (`elasticsearch.exceptions.SerializationError: ({'my_field': ['value1', 'value2', 'value3']}, TypeError("Unable to serialize ['value1', 'value2', 'value3'] (type: <class 'elasticsearch_dsl.utils.AttrList'>)",)`).  
As I understand it, it happens since when affecting attribute `my_field` with the `AttrList` the actual list value (`_l_`)  is not extracted in `_d_`. 
Since an ES document with a list in a field is automagically deserialized as an `AttrList` in the `Hit` object, not being able to directly affect a field in a `DocType` with the value of another field in a `Hit` document seems not like a desirable behaviour. And it makes for a very nasty exception to understand (took me a while to understand what was happening).

**Comment by HonzaKral** [comment]
This should now work, however there are still caveats:

It only works for fields that are in an object that has a mapping - the top level object always has mappings (so your case will work) but if you'd like to do something like `blog_post.comments.tags = AttrList([...])` it will only work if `comments` are being properly defined as either a `Nested` or `Object` field.

Also you can avoid this problem by creating the low level client via the `connections` object which will register our custom serializer which recognizes `AttrDict` and `AttrList` classes.

Hope this helps!

**Comment by andreip** [comment]
Thanks for the fix. I too was reassigning the `AttrList` field in the init function of a `DocType` and doing `json.dumps(doc.to_dict())` would fail in a similar way `TypeError: Object of type 'AttrList' is not JSON serializable`.

Wrote a test to file an issue for 5.3.0 but then found it was fixed in 5.4.0 and ended up here. 👍

**Comment by littlehome-eugene** [comment]
I'm confused with this

https://github.com/elastic/elasticsearch-dsl-py/issues/95 says List is the default and suggest the following format

```
class Bog(DocType):

    tags = Keyword()
```

The comments here mentions, although I think it deals with the case where Blog has Comment and when Comment has Tag list, (as opposed to Blog has Tag)

> it will only work if comments are being properly defined as either a Nested or Object field.


Should I have something like the following?

class Bog(DocType):

    tags = Object(Keyword())


`
TypeError: ['abc', 'sns'] is not JSON serializable`
